master/internal/stream/utils.go

```diff
@@ -0,0 +1,63 @@
+package stream
```

Comment:
nit: I'm not the biggest fan of having production utilities and testing utilities in the same file, do you think it be worthwhile to move `splitDeletionsAndUpserts` into `system_intg_test.go`?

Reply:
yeah I think that's a better move - though if we end up adding more utils, I'd be supportive of a test_utils file